src/java/picard/sam/PositionBasedDownsampleSam.java

```diff
@@ -0,0 +1,361 @@
+ * <p/>
+ * Finally, the code has been designed to simulate sequencing less as accurately as possible, not for getting an exact downsample fraction.
+ * In particular, since the reads may be distributed non-evenly within the lanes/tiles, the resulting downsampling percentage will not be accurately
+ * determined by the input argument PROBABILITY. One should re-MarkDuplicates after downsampling in order to "expose" the duplicates who's representative has
+ * been downsampled away.
+ *
+ * @author Yossi Farjoun
+ */
+@CommandLineProgramProperties(
+        usage = "Class to downsample a BAM file while respecting that we should either get rid\n" +
+                "of both ends of a pair or neither end of the pair. In addition, this program uses the read-name \n" +
+                "and extracts the position within the tile from whence the read came from. THe downsampling is based on this position. \n" +
+                "results with the exact same input will produce the same results.\n" +
+                "\n" +
+                "<it>Note 1</it>: This is technology and read-name dependent. if your read-names do not have coordinate information, or if your\n" +
+                "BAM contains reads from multiple technologies (flowcell versions, sequencing machines) this will not work properly. \n" +
+                "This has been designed with Illumina MiSeq/HiSeq in mind.\n" +
+                "<it>Note 2</it>: The downsampling is _not_ random. It is deterministically dependent on the position of the read within its tile." +
+                "<it>Note 3</it>: Downsampling twice with this program is not supported." +
+                "<it>Note 4</it>: You should MarkDuplicates after downsampling." +
+                "\n" +
+                "Finally, the code has been designed to simulate sequencing less as accurately as possible, not for getting an exact downsample fraction. " +
+                "In particular, since the reads may be distributed non-evenly within the lanes/tiles, the resulting downsampling percentage will not be accurately" +
+                "determined by the input argument PROBABILITY.",
+        usageShort = "Down-sample a SAM or BAM file to retain a subset of the reads based on the reads location in each tile in the flowcell.",
+        programGroup = SamOrBam.class
+)
+public class PositionBasedDownsampleSam extends CommandLineProgram {
+
+    @Option(shortName = StandardOptionDefinitions.INPUT_SHORT_NAME, doc = "The input SAM or BAM file to downsample.")
+    public File INPUT;
+
+    @Option(shortName = StandardOptionDefinitions.OUTPUT_SHORT_NAME, doc = "The output, downsampled, SAM or BAM file to write.")
+    public File OUTPUT;
+
+    @Option(shortName = StandardOptionDefinitions.PROBABILITY_SHORT_NAME, doc = "The (approximate) probability of keeping a read, between 0 and 1.")
+    public double PROBABILITY = 1;
```

Comment:
Is `PROBABILITY` really the right terminology for what seems to be a deterministic fraction of the tile area?